src/Planner/PlannerJoinTree.cpp

```diff
@@ -1102,6 +1139,17 @@ JoinTreeQueryPlan buildQueryPlanForTableExpression(QueryTreeNodePtr table_expres
                 }
 
                 auto & alias_column_expressions = table_expression_data.getAliasColumnExpressions();
+                if (has_table_virtual_column && query_plan.isInitialized())
```

Comment:
This should be done only in case `from_stage == FetchColumns`.
Otherwise, we should not expect the column in the header.
Example: `select sum(x) from remote('127.0.0.{1,2}', ...) where _table = 'my_table' group by y`
The query would be executed up to `WithMergableStage` with a header like `y, AggregateFunctionState(sum, x)` (and _table column should be processed by the storage itself).

Reply:
@KochetovNicolai Added a check for from_stage == QueryProcessingStage::FetchColumns.